android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java

```diff
@@ -149,13 +150,19 @@ protected void setShouldIgnoreSelectionChange(boolean ignore) {
 
   protected boolean updateText(String text) {
     boolean result = false;
+    JSONObject reg = new JSONObject();
     String kmText = "";
     if (text != null) {
-      kmText = text.toString().replace("\\", "\\u005C").replace("'", "\\u0027").replace("\n", "\\n");
+      // Use JSON to handle passing string to Javascript
+      try {
+        reg.put("text", text.toString());
+      } catch (JSONException e) {
+        KMLog.LogException(TAG, "", e);
+      }
```

Comment:
We shouldn't be needing to catch this, I can't see any scenarios in which this could happen.

Reply:
The Gradle build fails if I remove these:

```
error: unreported exception JSONException; must be caught or declared to be thrown
       reg.put("text", text.toString());
```

Reply:
ah yeah okay

Reply:
Yep, that's Java for you.  Thou shalt declare all unhandled exceptions, one way or another.